### PR TITLE
store-symbols-local-storage

### PR DESCRIPTION
**Pull request by pranjaljately**

### CODE REVIEW

File: src/routes/index.svelte [hunk 1]
```diff
@@ -73,23 +50,82 @@
     let loading = false
     let highlightedSymbolIndex = null
 
+    let isDocumentReady = false
+
     onMount(() => {
+        let DEFAULT_OPTIONS = []
```
Comment (zubhav) [inline src/routes/index.svelte:56]: Change variable name to `selectedOptions` or something as it is no longer default options and no longer a constant

File: src/routes/index.svelte [hunk 2]
```diff
@@ -73,23 +50,82 @@
...
+
+        if (typeof document !== 'undefined') {
```
Comment (zubhav) [inline src/routes/index.svelte:58]: Is there ever an instance where document is undefined within `onMount`? I would've assumed not

File: src/routes/index.svelte [hunk 3]
```diff
@@ -73,23 +50,82 @@
...
+            symbolList = symbolsString.split(',')
+        }
+
+        if (symbolList && symbolList.length > 0) {
+            return symbolList.map(symbol => {
+                return { symbol }
+            })
+        }
+
+        const options = [
```
Comment (zubhav) [inline src/routes/index.svelte:85]: Can name this DEFAULT_OPTIONS and add it to the top of the file as a constant

File: src/routes/index.svelte [hunk 4]
```diff
@@ -73,23 +50,82 @@
...
+        const storedSymbols = localStorage.getItem(LS_SYMBOLS)
+
+        if (symbols) {
+            symbolList = symbols.split(',')
+        } else if (storedSymbols) {
+            const symbolsString = storedSymbols
+            symbolList = symbolsString.split(',')
+        }
+
+        if (symbolList && symbolList.length > 0) {
```
Comment (zubhav) [inline src/routes/index.svelte:79]: You can declare `let symbolList = []` and remove `symbolList &&` from this if statement

File: src/routes/index.svelte [hunk 5]
```diff
@@ -73,23 +50,82 @@
...
+
+        DEFAULT_OPTIONS = getDefaultOptions(symbols)
         fetchStockData(DEFAULT_OPTIONS)
     })
 
+    const getDefaultOptions = symbols => {
+        let symbolList
+        const storedSymbols = localStorage.getItem(LS_SYMBOLS)
+
+        if (symbols) {
```
Comment (zubhav) [inline src/routes/index.svelte:72]: If we have query parameters, we should replace what is in local storage

File: src/routes/index.svelte [hunk 6]
```diff
@@ -73,23 +50,82 @@
...
+            symbolList = symbols.split(',')
+        } else if (storedSymbols) {
+            const symbolsString = storedSymbols
```
Comment (zubhav) [inline src/routes/index.svelte:75]: You can remove this line and do:

`symbolList = storedSymbols.split(',')`

File: src/routes/index.svelte [hunk 7]
```diff
@@ -73,23 +50,82 @@
...
+            },
+        ]
+
+        return options
+    }
+
+    const getDefaultPriceOption = price => {
+        const storedPrice = localStorage.getItem(LS_PRICE)
+
+        if (price && priceOptionExists(price)) {
```
Comment (zubhav) [inline src/routes/index.svelte:106]: Same here as above. Store it in local storage if it comes in as a query parameter.

File: src/routes/index.svelte [hunk 8]
```diff
@@ -73,23 +50,82 @@
...
+            return price
+        }
+
+        if (storedPrice && priceOptionExists(storedPrice)) {
+            return storedPrice
+        }
+
+        const priceOption = 'close'
```
Comment (zubhav) [inline src/routes/index.svelte:114]: DEFAULT_PRICE_OPTION = 'close' and can move this to the top

File: src/routes/index.svelte [hunk 9]
```diff
@@ -215,12 +270,23 @@
...
     }
+
+    $: {
+        selectedPriceOption &&
+            isDocumentReady &&
+            saveSelectedPrice(selectedPriceOption)
+    }
+
+    $: {
+        history, isDocumentReady && saveSymbols(history)
```
Comment (zubhav) [inline src/routes/index.svelte:288]: Shouldn't this be `history && isDocumentReady && ...`

File: src/routes/index.svelte [hunk 10]
```diff
@@ -215,12 +270,23 @@
...
             selectedPriceOption &&
+                isDocumentReady &&
                 updateQueryParams(history, selectedPriceOption)
     }
 
     $: {
         history.size && resetHighlightedSymbol()
     }
+
+    $: {
```
Comment (zubhav) [inline src/routes/index.svelte:281]: 👍

File: src/routes/index.svelte [hunk 11]
```diff
@@ -104,6 +140,25 @@
         }
     }
 
+    const saveSymbols = history => {
```
Comment (zubhav) [inline src/routes/index.svelte:143]: `saveSelectedSymbols`

File: src/routes/index.svelte (hunk 5 above)
Comment (pranjaljately) [reply src/routes/index.svelte:72]: This gets handled by the reactive declaration when history is updated.

File: src/routes/index.svelte (hunk 7 above)
Comment (pranjaljately) [reply src/routes/index.svelte:106]: Same as previous - gets handled by the reactive declaration when selectedPriceOption is reassigned.

File: src/routes/index.svelte [hunk 12]
```diff
@@ -104,6 +140,25 @@
...
+        const currentSymbols = Array.from(history.keys())
+        let symbolList = ''
+
+        if (currentSymbols.length > 0) {
+            symbolList = currentSymbols.join(',')
+            localStorage.setItem(LS_SYMBOLS, symbolList)
```
Comment (zubhav) [inline src/routes/index.svelte:149]: In another PR, might be worth creating a localStorage.utils.js with the following:

`const setLocalStorageItem = (key, value) => localStorage.setItem(key, value);

`const getLocalStorageItem = (key, value) => localStorage.getItem(key, value);`

So we can do this:
`setLocalStorageItem(LS_SYMBOLS, symbolList)`

Comment (pranjaljately) [reply src/routes/index.svelte:146]: Sure 👍

**Pull request merged**